modules/bedToNrf.cfmod.pl

```diff
@@ -0,0 +1,112 @@
+#                                                                        #
+# This file is part of Cluster Flow.                                     #
+#                                                                        #
+# Cluster Flow is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+        'cores'         => '1',
+        'memory'        => '1G',
+        'modules'       => '',
+        'time'          => sub {
+
+                my $cf = $_[0];
+                my $num_files = $cf->{'num_starting_merged_aligned_files'};
+                # Default to 1 if none were parsed for whatever reason
+                $num_files = ($num_files > 0) ? $num_files : 1;
+                # Ususally takes around 5 mins per file. Set to 30 mins just to be sure.
+                return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 30);
+        }
+);
+
+
+# The help text
+my $helptext = "".("-"x15)."\n BedToNrf\n".("-"x15)."\n
+Takes bed file and computes the NRF (non-redundant fraction).
+Uses code taken from https://github.com/mel-astar/mel-ngs/blob/master/mel-chipseq/chipseq-metrics/CalBedNrf.pl
+Output is a txt file with the total number of reads, the number of unique reads and the  NRF. 
+This file is named basename_nrf.txt.\n\n";
+
+# Start your engines...
+my %cf = CF::Helpers::module_start(\%requirements, $helptext);
+
+# Open up our run file in append mode
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Go through each file and run our command
+foreach my $file (@{$cf{'prev_job_files'}}){
+    # Start the clock...
+    my $timestart = time;
+
+    # Check that input is bed file
+    if(!($file =~ /\.bed$/)){
+	warn "\n###CF Error! bedToNrf failed: bed file expected, got $file\n\n";
+	next;
+    }
+    
+    # Generate a nice output file name
+    my $output_fn = $file."_nrf.txt";
+    
+    # Go through bed file and compute NRF (non redundant fraction) of reads.
+    open(IN,"<$file")||die "###CF Error! bedToNrf failed: $!\n";
```

Comment:
Maybe a more explicit error message here, notably mentioning which file couldn't be loaded?

```perl
open(IN,"<$file") || die "###CF Error! bedToNrf could not load input file '$file': $!\n";
```